src/main/java/de/tum/cit/aet/artemis/programming/service/localvc/ArtemisGitServletService.java

```diff
@@ -58,6 +58,7 @@ public void init() {
         this.setReceivePackFactory((request, repository) -> {
             ReceivePack receivePack = new ReceivePack(repository);
             // Add a hook that prevents illegal actions on push (delete branch, rename branch, force push).
+            // the user inside the request is always null here
```

Comment:
Just a question for understanding, why do we not directly call the method with null then or why do we even have the parameter?

Reply:
In the future there might be situations where this is not null any more, and could maybe be used